dev.c

```diff
@@ -0,0 +1,66 @@
+/*
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+#include <defs.h>
+#include <boot.h>
+#include <types.h>
+#include <pci.h>
+#include <dev.h>
+
+u32 dev_locate(void)
+{
+	return pci_locate(DEV_PCI_BUS,
+	                  PCI_DEVFN(DEV_PCI_DEVICE, DEV_PCI_FUNCTION));
+}
+
+u32 dev_read(u32 dev, u32 function, u32 index)
+{
+    u32 value;
+
+    pci_write(0, DEV_PCI_BUS,
+        PCI_DEVFN(DEV_PCI_DEVICE,DEV_PCI_FUNCTION),
+        dev + DEV_OP_OFFSET,
+        4,
+        (u32)(((function & 0xff) << 8) + (index & 0xff)) );
```

Comment:
It seems you're going for Xen style elsewhere, but there's a stray space here.

Also, you'd reduce the churn substantially if you fixed up the style in patch 2 for the logic you pull out of other files, because then you're not double patching it.

Reply:
I guess my thinking is a little counter to that. Since I knew the next change was to do whitespace, I would do it all at once. But I get your point that I move code and then touch it again for style.